### ISSUE TITLE
Several issues parsing patterns

### ISSUE THREAD

**Issue by diegodorado** [tidalcycles/Tidal#433]
I had many issues parsing patterns, used with `scale` function ( not sure if scale has anything to do though)

Here some simple examples of what failed trying to isolate the error. 

```haskell

-- this works
d1 $ note (scale "minor" "[[-1 0 ~]*8]/3") # s "midi"

-- this doesnt take into account the /3
d1 $ note (scale "minor" "[-1 0 ~]*8/3") # s "midi"

-- this breaks
d1 $ note (scale "minor" "[-1 0 ~]*(8/3)") # s "midi"
-- *** Exception: Syntax error in sequence:"[-1 0 ~]*(8/3)" -- ^ unexpected "(" expecting white space, "-", "+", natural, "[", "{" or "<"


-- this breaks
d1 $ note (scale "minor" "{-1 0 ~}%8") # s "midi"
-- *** Exception: Syntax error in sequence:
-- "{-1 0 ~}%8" ^ unexpected end of input expecting end of comment 

```

**Comment by yaxu** [comment]
Interesting, it seems to be treating `{-` as the start of a comment

This probably works: `d1 $ note ( scale "aeolian" $ "{ -1 0 ~}") # s "midi"`

**Comment by diegodorado** [comment]
fixed on 1.0.2
